website/templates/project/retract_registration.mako

```diff
@@ -0,0 +1,69 @@
+<%inherit file="project/project_base.mako"/>
+<%def name="title()">Retract Registration of Component</%def>
+
+<legend class="text-center">Retract Registration</legend>
+
+% if schema:
+    <%include file="metadata/register_${str(metadata_version)}.mako" />
+% else:
+
+    <div id="registrationRetraction" class="col-md-6 col-md-offset-3">
+        <form id="registration_retraction_form" role="form">
+
+            <div class="form-group">
+                <label class="control-label">Please provide your justification for retracting this registration.</label>
+                <textarea
+                        class="form-control"
+                        data-bind="textInput: justification"
+                        id="justificationInput"
+                        >
+
+                </textarea>
+            </div>
+
+            <hr />
+
+            <div class="form-group">
+                <label class="control-label">Type '<span data-bind="text: registrationTitle"></span>' if you are sure you want to continue.</label>
+                <textarea
+                        class="form-control"
+                        data-bind="textInput: confirmationText"
+                        >
+
+                </textarea>
+            </div>
+            <button type="submit" class="btn btn-danger" data-bind="click: submit, visible: true">Retract Registration</button>
```

Comment:
Neither. Allow the button to be clicked, but show a proper error message if the data are invalid. See https://github.com/CenterForOpenScience/osf.io/issues/1035